test/unit/tap-reporter.js

```diff
@@ -0,0 +1,87 @@
+/* eslint-env mocha */
+
+var chai = require('chai')
+var sinon = require('sinon')
+var EventEmitter = require('events').EventEmitter
+var JsReporters = require('../../dist/js-reporters.js')
+var data = require('./data.js')
+var expect = chai.expect
+
+chai.use(require('sinon-chai'))
+
+describe('Tap reporter', function () {
+  var emitter
+
+  before(function () {
+    emitter = new EventEmitter()
+    var reporter = new JsReporters.TapReporter(emitter)
+  })
+
+  it('should output the TAP header', function () {
+    var spy = sinon.stub(console, 'log')
+
+    emitter.emit('runStart', {})
+    console.log.restore()
+
+    expect(spy).to.have.been.calledOnce
+  })
+
+  it('should output ok for a passing test', function () {
+    var spy = sinon.stub(console, 'log')
```

Comment:
I must do this stubbing in each test and then the restoring below, because otherwise Mocha's reporter output will be also captured and will not be outputted to the console.